extensions/amp-timeago/0.2/amp-timeago.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {Timeago} from './timeago';
+import {isExperimentOn} from '../../../src/experiments';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-timeago';
+
+class AmpTimeago extends PreactBaseElement {
+  /** @override */
+  init() {
+    return {'init': this.element.textContent};
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
```

Comment:
For components it's typical to do this in the `buildCallback`, or if it is a smaller feature to the component (e.g. via attribute), by gating the functionality at read. Seeing as `buildCallback` isn't used and there is no need for a constructor block in this case, `isLayoutSupport` is a good location since it is always guaranteed to be called. I can't take credit though since @jridgewell did this for amp-date-display in #25969 :)